server.js

```diff
@@ -82,40 +30,61 @@ app.get("/users/:Id", (req, res) => {
     }
 });
 
-//now i can update any parameter of users, ot a total user if i change a little bit at line 58
-app.put("/users/:Id", (req, res) => {
-    const jsonData = require('./user.json');
-    const updatedObject = req.params.Id;
-    let indexPut = jsonData.findIndex(updatedObject => {
-        return updatedObject.name === `${req.params.Id}`;
-    });
-    if (indexPut > -1) {
-        jsonData[indexPut] = req.body;
-        res.status(200).json
+// Validate and add users to the user.json, if the users file does not exist, creates the file with append.
+expressServer.post("/users", (req, res) => {
+    const postValidation = (validationUserPost(req.body));
+    if (postValidation) {
+        addUserPost(req.body);
+        res.status(201).json({
+            message: `The user ${req.body.name} is succesfully created!`
+        });
+    } else {
+        res.status(403).json({
+            message: `The user ${req.params.Id} cannot be validated!`
+        })
+    }
+});
+
+// The first if add a user and creates the file if the user.json file does not exist yet.
+expressServer.put("/users/:Id", (req, res) => {
+    const putValidation = putUserbyId(req.body, req.params.Id);
+    const putValidationAsPost = (validationUserPost(req.body));
+    if(putValidationAsPost !== true) {
+        res.status(404).json({
```

Comment:
404 is for "not found" errors. Normally you use something else, like 422.

Reply:
yes it is 422 indeed, updated!